### PR TITLE
polok task/graphql appsync dynamodb resolver

### PR DESCRIPTION
**Pull request by codefresher32**
Fixes #

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [x] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
- Query using graphql appsync dynamodb resolver

## What was done
- Added a appsync resource
- Used dynamodb resolver in the appsync
- Added lambda resolver in the appsync

## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

**Review by rkzico21 — CHANGES_REQUESTED** [review]
Please address the feedback below. Also add a version of query that uses lambda as resolver.

File: schema.graphql [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+type Query {
+  getActivityInfo(CUSTOMER: String!, USER: String): [userActivity]
+}
+
+schema {
+	query: Query
+	mutation: Mutation
+}
```
[the remark this answers is not in the document]
Comment (codefresher32) [reply schema.graphql]: Used PascelCase for type and input name and camelCase for property name...

[the remark this answers is not in the document]
Comment (codefresher32) [reply schema.graphql]: Added lambda resolver...

File: schema.graphql [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+type Mutation {
+  AddUserInfo(input: AddUserInput!): UserActivity
+  AddUserInfoByLambda(input: AddUserInput!): UserActivity
+}
+
+type Query {
+  GetActivityInfo(customer: String!, user: String): [UserActivity]
+  GetActivityInfoByLambda(customer: String!, user: String): [UserActivity]
```
Comment (rkzico21) [inline schema.graphql:23]: Instead of providing number of parameters to the query define an input name GetActivityInput

File: handler.js [hunk 3]
```diff
@@ -120,6 +120,29 @@ module.exports.hello = async (event) => {
...
+      if (!isValidActivityData(event)) {
+        const message = 'Please Provide Proper Activity in Appsync';
+        return responseMessage(400, JSON.stringify(message));
+      }
+      try {
+        await insertIntoDynamoBD(event);
+      } catch (error) {
+        return responseMessage(500, JSON.stringify(error.message));
+      }
+    }
```
Comment (rkzico21) [inline handler.js:145]: There is too much if/else nesting in the code block. Please reduce nesting by splitting this portion into couple of functions for query and mutation.

**Review by rkzico21 — CHANGES_REQUESTED** [review]
Please have a look at the feedback.

File: schema.graphql (hunk 2 above)
Comment (codefresher32) [reply schema.graphql:23]: Created input type named GetActivityInput....

File: handler.js (hunk 3 above)
Comment (codefresher32) [reply handler.js:145]: Divided the query and mutation part in the function...

File: handler.js [hunk 4]
```diff
@@ -285,3 +291,40 @@ const isUserExist = (body) => {
...
+
+const isAppSyncQueryEvent = (event) => {
+  return event.eventName && event.eventName === 'query';
+};
+
+const isAppSyncMutationEvent = (event) => {
+  return event.eventName && event.eventName === 'mutation';
+};
+
+const queryIntoDynamoDBUsingAppSync = async (event) => {
```
Comment (rkzico21) [inline handler.js:307]: It would be more appropriate to rename the function getActivities

File: handler.js [hunk 5]
```diff
@@ -285,3 +291,40 @@ const isUserExist = (body) => {
...
+  }
+  try {
+    const data = await queryIntoDynamoBD(event);
+    return data;
+  } catch (error) {
+    return responseMessage(500, JSON.stringify(error.message));
+  }
+};
+
+const mutationIntoDynamoDBUsingAppSync = async (event) => {
```
Comment (rkzico21) [inline handler.js:320]: Rename it to createActivity(or addActivity)

**Review by rkzico21 — CHANGES_REQUESTED** [review]
Some minor feedback.  Please have a look.

File: handler.js (hunk 4 above)
Comment (codefresher32) [reply handler.js]: renamed the query function...

File: handler.js (hunk 5 above)
Comment (codefresher32) [reply handler.js]: renamed the mutation function....

**Comment by codefresher32** [comment]
Please review this pr....

**Review by rkzico21 — APPROVED** [review]
Looks good.

**Pull request merged**